### PR TITLE
Authentication added to api and query addition

### PR DESCRIPTION
**Pull request by Gaurav918**
added authentication in some of the apis and used email, userid directly from the session using authentication.

### CODE REVIEW

File: src/main/java/com/voice/dbRegistration/dao/DevoteeInfoDao.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import com.voice.dbRegistration.model.DevoteeInfo;
 
 public interface DevoteeInfoDao extends JpaRepository<DevoteeInfo, String> {
+    @Query(value = "select * FROM DevoteeInfo d where d.connectedTo like '%guru%'")
```
Comment (Vedgaurav) [inline src/main/java/com/voice/dbRegistration/dao/DevoteeInfoDao.java:12]: We don't need "d" here. We are not using any join here

File: src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java [hunk 2]
```diff
@@ -43,15 +52,16 @@ private DevoteeInfo encryptData(DevoteeInfo input) {
     }
 
     @PostMapping("/doesUserExist")
-    public List<DevoteeInfo> doesExist(@RequestBody Map<String, String> input) {
-        String email = input.get("email");
-        return devoteeInfoDao.findAllByEmail(email);
+    public ResponseEntity<List<DevoteeInfo>> doesExist(Authentication authentication) {
+        Optional<UserAuth> user=userAuthService.getUserAuthFromAuthentication(authentication);
+        return user.map(userAuth -> ResponseEntity.ok(devoteeInfoDao.findAllByEmail(userAuth.getUserEmail()))).orElseGet(() -> ResponseEntity.status(HttpStatus.UNAUTHORIZED).build());
+
```
Comment (Vedgaurav) [inline src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java:58]: We should get only one user from db query. So why are we returning List<DevoteeInfo>?

File: src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java [hunk 3]
```diff
@@ -43,15 +52,16 @@ private DevoteeInfo encryptData(DevoteeInfo input) {
...
     }
 
     @PostMapping("/fetchAllDepById/{userId}")
```
Comment (Vedgaurav) [inline src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java:61]: As we are not using this userId from request we can remove it

File: src/main/java/com/voice/dbRegistration/dao/DevoteeInfoDao.java (hunk 1 above)
Comment (Gaurav918) [reply src/main/java/com/voice/dbRegistration/dao/DevoteeInfoDao.java:12]: applied changes

File: src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java (hunk 2 above)
Comment (Gaurav918) [reply src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java:58]: applied changes

File: src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java (hunk 3 above)
Comment (Gaurav918) [reply src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java]: applied changes

**Review by Vedgaurav — APPROVED** [review]

**Pull request merged**